AnimationCodelab/finished/src/main/java/com/example/android/codelab/animation/ui/home/Home.kt

```diff
@@ -0,0 +1,702 @@
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.composed
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.vector.ImageVector
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.input.pointer.positionChange
+import androidx.compose.ui.input.pointer.util.VelocityTracker
+import androidx.compose.ui.res.stringArrayResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.example.android.codelab.animation.R
+import com.example.android.codelab.animation.ui.AnimationCodelabTheme
+import com.example.android.codelab.animation.ui.amber600
+import com.example.android.codelab.animation.ui.green300
+import com.example.android.codelab.animation.ui.green800
+import com.example.android.codelab.animation.ui.purple100
+import com.example.android.codelab.animation.ui.purple700
+import kotlinx.coroutines.coroutineScope
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import kotlin.math.absoluteValue
+import kotlin.math.roundToInt
+
+/**
+ * Shows the entire screen.
+ */
+@Composable
+fun Home() {
+    // String resources.
+    val allTasks = stringArrayResource(R.array.tasks)
+    val allTopics = stringArrayResource(R.array.topics).toList()
+
+    // The index of the selected tab.
+    var currentTab by remember { mutableStateOf(0) }
+
+    // True if the whether data is currently loading.
+    var weatherLoading by remember { mutableStateOf(false) }
+
+    // Holds all the tasks currently shown on the task list.
+    val tasks = remember { mutableStateListOf(*allTasks) }
+
+    // Holds the topic that is currently expanded to show its body.
+    var expandedTopic by remember { mutableStateOf<String?>(null) }
+
+    // True if the message about the edit feature is shown.
+    var editMessageShown by remember { mutableStateOf(false) }
+
+    // Starts loading weather data. This takes 3 seconds.
+    suspend fun loadWeather() {
+        if (!weatherLoading) {
+            weatherLoading = true
+            delay(3000L)
+            weatherLoading = false
+        }
+    }
+
+    // Shows the message about edit feature.
+    suspend fun showEditMessage() {
+        if (!editMessageShown) {
+            editMessageShown = true
+            delay(3000L)
+            editMessageShown = false
+        }
+    }
+
+    LaunchedEffect(Unit) {
+        loadWeather()
+    }
+    val lazyListState = rememberLazyListState()
+
+    // The background color. The value is changed by the current tab.
+    val backgroundColor by animateColorAsState(if (currentTab == 0) purple100 else green300)
```

Comment:
Could this use theme colors instead? e.g. switching between `MaterialTheme.primary` and `MaterialTheme.secondary`?

Reply:
That's how things should be in real life, but I didn't want to add prerequisite for knowledge about themes. People might try this codelab before the theme one.